zio-http/src/main/scala/zhttp/service/server/content/handlers/ServerResponseHandler.scala

```diff
@@ -111,6 +112,11 @@ private[zhttp] trait ServerResponseHandler[R] {
       case HttpData.JavaFile(unsafeGet) =>
         unsafeWriteFileContent(unsafeGet())
         releaseAndRead(jReq)
+
+      case d: HttpData.UnsafeAsync =>
+        rt.unsafeRun(ctx) {
+          writeStreamContent(d.toByteBufStream(ByteBufConfig())).ensuring(ZIO.succeed(releaseAndRead(jReq)))
+        }
```

Comment:
```suggestion

      case HttpData.UnsafeAsync(unsafeRun) =>
        unsafeRun {
          _ => c => ctx.writeAndFlush(c.content): Unit
        }
```

I think something simpler like above should suffice for this usecase.

Reply:
This code does not work for me.

Reply:
This works, but I believe stream is better:
```scala
      case HttpData.UnsafeAsync(unsafeRun) =>
        unsafeRun { ch => msg =>
          runtime.unsafeRun(ctx)(ChannelFuture.unit(ctx.writeAndFlush(msg.content)))
          if (msg.isLast)
            runtime.unsafeRun(ctx)(ChannelFuture.unit(ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT)))
          else ch.read()
        }
```

Reply:
You are right. There is a missing `ctx.read()` call.
```scala
      case HttpData.UnsafeAsync(unsafeRun) =>
        unsafeRun {
          _ => { c => 
                 ctx.writeAndFlush(c.content)
                 ctx.read(): Unit // This was missing.
               }
        }
```

Reply:
This code fails all my tests (even the one with `HttpData.empty`).

Reply:
This one works, though I'm not sure if it is correct:
```scala
        unsafeRun { ch => msg =>
          ctx.writeAndFlush(msg.content).addListener { (_: Future[_]) =>
            if (msg.isLast) ctx.writeAndFlush(LastHttpContent.EMPTY_LAST_CONTENT): Unit
            else ch.read()
          }: Unit
        }
```

Reply:
See the new commit